src/PubSubClient.cpp

```diff
@@ -416,20 +420,22 @@ uint16_t PubSubClient::readPacket(uint8_t* lengthLength) {
         }
     }
 
-    for (uint16_t i = start;i<length;i++) {
+    uint32_t idx = len;
+    for (uint32_t i = start;i<length;i++) {
         if(!readByte(&digit)) return 0;
         if (this->stream) {
-            if (isPublish && len-*lengthLength-2>skip) {
+            if (isPublish && idx-*lengthLength-2>skip) {
```

Comment:
I know it wasn't in the original code, but shouldn't here be some kind of braces be added? At least for readability.
Also the part right from the && is now using 32, 8 and 16 bit values.
A `static_cast` would be a good indication someone at least thought about it, but I guess the `skip` should also be a 32 bit value if you also switch to 32 bit for the max. message length.  (not sure if that's possible with MQTT though)

Reply:
Those are okay changes but I think they're pretty minor and should be made onto the main branch, with the non-blocking changes from this branch manually merged into that as well, if found worthy. It's been too long since I submitted this PR, there's apparently a bunch of conflicts now and who knows how many tests failing. Spent too much time on this already, trying to sync up.